app.js

```diff
@@ -1,31 +1,23 @@
 const express = require("express")
 const {getTopics, getApi, getArticles} = require(`${__dirname}/controllers/topics.controller.js`)
+const {handlePsqlErrors, handleCustomErrors, handleServerErrors, handleInvalidEndpoints} = require(`${__dirname}/controllers/errors.controller.js`)
 
 const app = express()
 
 app.get("/api", getApi)
 
 app.get("/api/topics", getTopics)
 
+app.get("/api/articles", getArticles)
+
 app.get("/api/articles/:id", getArticles)
 
-app.use((err,request,response,next)=>{
-    if(err.code==='22P02'){
-      response.status(400).send({msg:'bad request'})
-    }
-    next(err)
-  })
-
-app.use((err, request, response, next) => {
-	if (err.status && err.msg) {
-		response.status(err.status).send({ msg: err.msg });
-	} else {
-		response.status(500).send({ msg: "Internal Server Error" });
-	}
-});
-
-app.all("/*", (request, response)=>{
-    response.status(404).send({msg:"Not found"})
-})
+app.use(handlePsqlErrors)
+
+app.use(handleCustomErrors)
+
```

Comment:
Nice to see this extracted out. Errors aren't really part of controllers though. I would recommend for a tidy structure and easy to find, a folder just for errors. While only one file, it has a clear location for you and anyone interacting with your project (like me)